Make HAproxy config more generic

oybed opened the pull request:
### What does this PR do?
This PR makes the HAproxy configuration more generic and allow for better control over the configuration. It also streamlines some of the process and speeds up execution. 

### How should this be tested?
Use with a HAproxy setup with the proper inventory (see README)

### Is there a relevant Issue open for this?
N/A

### Other Relevant info, PRs, etc.
N/A

### People to notify
cc: @redhat-cop/infra-ansible

sabre1041 commented on roles/haproxy-config/templates/lb_backend.j2:
```diff
@@ -0,0 +1,33 @@
+
+{% if ((be.lb_ssl_enabled is defined and be.lb_ssl_enabled) or
+       (be.lb_ssl_enabled is undefined and be.lb_port|string == '443')) %}
+{% set lb_ssl_enabled = True %}
```
@oybed there should be an else clause to ensure the variable is always set based on the conditional

```
{% if ((be.lb_ssl_enabled is defined and be.lb_ssl_enabled) or
       (be.lb_ssl_enabled is undefined and be.lb_port|string == '443')) %}
{% set lb_ssl_enabled = True %}
{% else %}
{% set lb_ssl_enabled = False %}
{% endif %}
```

oybed commented:
@sabre1041 updated with a few changes:
```
- moved to a new sub-directory under roles
- restructured the configuration dict based on our convo
- updated playbook to truly use a temp file
- moving away from with_items per latest Ansible direction
```

tylerauerbeck commented:
@oybed @sabre1041 LGTM

sabre1041 commented:
@oybed Tested the changes and ran into the following error

```
TASK [keepalived : Populate the keepalived.conf file] ****************************************************************************************************************************************************************************************
fatal: [10.9.53.38]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'keepalived_nic' is undefined"}
```

sabre1041 commented:
@oybed that sounds like a good approach. LGTM
